woocommerce/woocommerce-ios
# [Manage Categories]Synchronize product categories

Refs https://github.com/woocommerce/woocommerce-ios/issues/2000

# Why

Following #2043 where an action to retrieve product categories from a remote source was introduced.
This PR modifies that action to store such product categories in our storage layer.

All this efforts are made to power the ProductCategory Selector screen in a following PR.
![Screen Shot 2020-04-02 at 4 47 50 PM](https://user-images.githubusercontent.com/562080/78303135-b75d8d00-7501-11ea-8d17-6afb877f445f.png)


# How
- Adds `SiteID` and `ParentID` to Networking.ProductCategory and `Storage.ProductCategory` - includes xcdatamodel migration to v27

- Update `loadProductCategory` method to not rely on `productID` as categories would be shared through several products.

- Update `ProductCategoryStore` to upset categories in our storage layer upon a successful `loadAllProductCategories` network response.

## Notes
- The big diff is due to the new `xcdatamodel` file.

# Testing Steps
- Open the app with a logged site in a state previous to this PR
- Open the app with a logged site with the code on this PR
- Make sure migration works and no information is lost 😇
